wandb/sdk/internal/sender.py

```diff
@@ -991,12 +997,56 @@ def finish(self):
             self._fs = None
 
     def _max_cli_version(self):
-        _, server_info = self._api.viewer_server_info()
+        server_info = self.get_server_info()
         max_cli_version = server_info.get("cliVersionInfo", {}).get(
             "max_cli_version", None
         )
         return max_cli_version
 
+    def get_viewer_server_info(self):
+        if not self._cached_server_info or not self._cached_viewer:
+            (
+                self._cached_viewer,
+                self._cached_server_info,
+            ) = self._api.viewer_server_info()
+
+    def get_viewer_info(self):
+        if not self._cached_viewer:
+            self.get_viewer_server_info()
+        return self._cached_viewer
+
+    def get_server_info(self):
+        if not self._cached_server_info:
+            self.get_viewer_server_info()
+        return self._cached_server_info
+
+    def get_local_info(self):
```

Comment:
See earlier comment. Might be a good idea to move this logic into internal_api. Ideally callers wouldn't have to reason about differing gql schema versions.

Reply:
Hmmm... there are 2 reasoning happening here.
The first if/else is the 2 different schemas, but the second if/else is reasoning whether the jobber/cron run yet or not (before it runs for the first time, should be every 24 hours, we get `None` for `latestLocalVersionInfo`)

Reply:
No need, it's better now. My comment was mostly about this caller needing to know about the server info introspection.

Reply:
Seems like the these changes break windows run... so i'm reverting them for now